crates/turborepo-lib/src/shim.rs

```diff
@@ -305,23 +323,30 @@ pub fn run() -> Result<Payload> {
     // global turbo having handled the inference. We can run without any
     // concerns.
 
-    if !args.has_json_flags() {
+    if !args.should_skip_updater() {
```

Comment:
Could we have a `should_check_for_update` and then keep this from being inverted?

Reply:
Good call - will do!